voting/src/general_meeting.rs

```diff
@@ -0,0 +1,448 @@
+    pub meeting_id: GeneralMeetingId,
+}
+
+/// Two type of TX: `CreateGeneralMeeting` is a SignedTransaction while `PublishResult` is a UserTransaction.
+impl Action for TxGeneralMeeting {}
+impl Action for TxPublishResult {}
+pub type MeetingOwnTransaction = crate::common::SignedTransaction<TxGeneralMeeting>;
+pub type PublishOwnTransaction = crate::common::UserTransaction<TxPublishResult>;
+
+enum TxResult {
+    GeneralMeeting(GeneralMeetingId),
+    Result,
+}
+
+impl Context {
+    /// TXs are executed here in this function.
+    /// Rigorous conditions are checked here for both TXs although they are checked in `check_transaction` function.
+    fn excute_tx(&mut self, transaction: &Transaction) -> Result<TxResult, ExecuteError> {
+        match transaction.tx_type() {
+            MEETING_TX_TYPE => {
+                let tx: MeetingOwnTransaction =
+                    serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+                tx.verify().map_err(|_| ExecuteError::InvalidSign)?;
+
+                let num_agendas = tx.tx.action.number_of_agendas;
+                let end_time = tx.tx.action.end_time;
+                let tallying_time = tx.tx.action.tallying_time;
+                let now = self.block_header.as_ref().unwrap().timestamp();
+                if now > end_time.time {
+                    return Err(ExecuteError::WrongEndTime)
+                }
+                if now > tallying_time.time {
+                    return Err(ExecuteError::WrongTallyingTime)
+                }
+                if tallying_time.time < end_time.time {
+                    return Err(ExecuteError::InvalidTallying)
+                }
+
+                let meeting = GeneralMeeting::new(end_time, tallying_time, num_agendas);
+                let key = meeting.id.clone();
+
+                self.storage_mut().set(key.as_ref(), serde_cbor::to_vec(&meeting).unwrap());
+
+                let vote_box = VoteBox::new(meeting.id.clone());
+                let box_key = crate::generate_voting_box_key(&meeting.id);
+                self.storage_mut().set(box_key.as_slice(), serde_cbor::to_vec(&vote_box).unwrap());
+                Ok(TxResult::GeneralMeeting(meeting.id))
+            }
+            PUBLISH_TX_TYPE => {
+                let tx: PublishOwnTransaction =
+                    serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+
+                let meeting_id = tx.action.meeting_id;
+                let key = crate::generate_voting_box_key(&meeting_id);
+                let vote_box: VoteBox = serde_cbor::from_slice(&self.storage().get(key.as_slice()).unwrap()).unwrap();
+                let mut vote_result = VoteResult::new();
+
+                let mut meeting: GeneralMeeting =
+                    serde_cbor::from_slice(&self.storage().get(meeting_id.as_ref()).unwrap())
+                        .map_err(|_| ExecuteError::InvalidMetadata)?;
```

Comment:
Could you split this line into multiple lines? Since there are multiple `unwrap`s, it is hard to read the role of unwraps.

I also changed the error handling. We should handle the case when the general meeting does not exist.
We don't need to handle the case when the serde deserialization fails. Since the state is only readable and writable by this module we can trust the deserialization.

```suggestion
                let mut meeting: GeneralMeeting = {
                    let bytes =
                        self.storage().get(meeting_id.as_ref()).ok_or_else(|| ExecuteError::GeneralMeetingNotFound)?;

                    serde_cbor::from_slice(&bytes)
                    .expect("General meeting is serialized by this code")
                };
```